optuna/trial/_frozen.py

```diff
@@ -409,47 +408,47 @@ def _set_values(self, v: Optional[Sequence[float]]) -> None:
     values = property(_get_values, _set_values)
 
     @property
-    def datetime_start(self) -> Optional[datetime.datetime]:
+    def datetime_start(self) -> datetime.datetime | None:
         return self._datetime_start
 
     @datetime_start.setter
-    def datetime_start(self, value: Optional[datetime.datetime]) -> None:
+    def datetime_start(self, value: datetime.datetime | None) -> None:
         self._datetime_start = value
 
     @property
-    def params(self) -> Dict[str, Any]:
+    def params(self) -> dict[str, Any]:
         return self._params
 
     @params.setter
-    def params(self, params: Dict[str, Any]) -> None:
+    def params(self, params: dict[str, Any]) -> None:
         self._params = params
 
     @property
-    def distributions(self) -> Dict[str, BaseDistribution]:
+    def distributions(self) -> dict[str, BaseDistribution]:
         return self._distributions
 
     @distributions.setter
-    def distributions(self, value: Dict[str, BaseDistribution]) -> None:
+    def distributions(self, value: dict[str, BaseDistribution]) -> None:
         self._distributions = value
 
     @property
-    def user_attrs(self) -> Dict[str, Any]:
+    def user_attrs(self) -> dict[str, Any]:
         return self._user_attrs
 
     @user_attrs.setter
-    def user_attrs(self, value: Dict[str, Any]) -> None:
+    def user_attrs(self, value: dict[str, Any]) -> None:
         self._user_attrs = value
 
     @property
-    def system_attrs(self) -> Dict[str, Any]:
+    def system_attrs(self) -> dict[str, Any]:
         return self._system_attrs
 
     @system_attrs.setter
     def system_attrs(self, value: Mapping[str, JSONSerializable]) -> None:
-        self._system_attrs = cast(Dict[str, Any], value)
+        self._system_attrs = cast(dict, value)
```

Comment:
This line slightly changes the type checking behavior.

The previous code was stricter than the change made by you.
For example, the last one did not allow `Dict[int, Any]`, because `cast` ensures that `value` is `Dict[str, Any]`.
However, the current code allows `Dict[Any, Any]`.

```suggestion
        self._system_attrs = cast(Dict[str, Any], value)
```

Reply:
Thanks, as you said, it is fixed in 9300fb5